packages/dds/sequence/docs/intervalCollectionQuerying.md

```diff
@@ -0,0 +1,240 @@
+	/** @internal */
+	add(interval: TInterval): void;
+	/** @internal */
+	remove(interval: TInterval): void;
+}
+```
+
+The 3 parts correspond to query APIs which depend on the minmax interval tree, the endpoint interval tree, and the id-to-interval map.
+
+Indexes can store data as is convenient, but must support addition and removal of that data.
+This keeps details about interval comparators being mutable internal to `LocalIntervalCollection`.
+
+Another nice feature of factoring the code this way is it increases testability: correctness for spatial aspects of each index can be verified using numeric intervals and without spinning up any DDS infrastructure.
+
+## Changing Public Query APIs
+
+After the internals more closely reflect our desired code factoring, the public API should follow.
+If we take the commit from the previous section, this would mean refactoring `IntervalCollection` to have a shape resembling
+
+```typescript
+export class IntervalCollection<TInterval extends ISerializableInterval> extends TypedEventEmitter<
+	IIntervalCollectionEvent<TInterval>
+> {
+	indexes: {
+		id: IIntervalIdIndex<TInterval>;
+		overlappingIntervals: IOverlappingIntervalsIndex<TInterval>;
+		matchingEndpoints: IMatchingEndpointsIndex<TInterval>;
+	};
+	add(start: number, end: number, intervalType: IntervalType, props?: PropertySet): TInterval;
+	addConflictResolver(conflictResolver: IntervalConflictResolver<TInterval>): void;
+	attachDeserializer(onDeserialize: DeserializeCallback): void;
+	change(id: string, start?: number, end?: number): TInterval | undefined;
+	changeProperties(id: string, props: PropertySet): void;
+	removeIntervalById(id: string): TInterval | undefined;
+}
+
+export interface IIntervalIdIndex<TInterval extends ISerializableInterval>
+	extends IntervalIndex<TInterval>,
+		Iterable<TInterval> {
+	getIntervalById(id: string): TInterval | undefined;
+}
+
+export interface IOverlappingIntervalsIndex<TInterval extends ISerializableInterval>
+	extends IntervalIndex<TInterval> {
+	CreateBackwardIteratorWithEndPosition(
+		endPosition: number,
+	): IntervalCollectionIterator<TInterval>;
+	CreateBackwardIteratorWithStartPosition(
+		startPosition: number,
+	): IntervalCollectionIterator<TInterval>;
+	CreateForwardIteratorWithEndPosition(
+		endPosition: number,
+	): IntervalCollectionIterator<TInterval>;
+	CreateForwardIteratorWithStartPosition(
+		startPosition: number,
+	): IntervalCollectionIterator<TInterval>;
+	findOverlappingIntervals(startPosition: number, endPosition: number): TInterval[];
+	gatherIterationResults(
+		results: TInterval[],
+		iteratesForward: boolean,
+		start?: number,
+		end?: number,
+	): void;
+}
+```
+
+Note: we probably don't want to check this step in directly,
+as ideally we should only need to break the public API once to reach our desired state.
+Making the set of indexes customizable will involve changing the type of "indexes" above.
+
+## Making supported indexes customizable
```

Comment:
i like that it opens to the door to this, but unless there is a need/ask, i'd wait to expose it